doc/CONFIGURATION.md

```diff
@@ -80,6 +81,23 @@ Here is an example least-privilege policy document to add to an IAM user or role
 
 Mountpoint also respects access control lists (ACLs) applied to objects in your S3 bucket, but does not allow you to automatically attach ACLs to objects created with Mountpoint. A majority of modern use cases in Amazon S3 no longer require the use of ACLs. We recommend that you keep ACLs disabled for your S3 bucket, and instead use bucket policies to control access to your objects.
 
+## Caching configuration
+
+Mountpoint now offers caching of metadata and object content allowing for reduced requests when reading files.
+This is particularly useful when reading the same files many times for the same Mountpoint filesystem.
+
+To enable caching, use the `--caching` command-line flag.
+This alone will enable caching of metadata using a default time-to-live (TTL) of 60 minutes.
+Caching of object/file content on disk can also be enabled
+by providing a caching location using the `--data-cache-directory <DIR>` command-line flag.
```

Comment:
Naming bikeshedding: `--cache-metadata`/`--cache-data <DIR>`?

And then mention that `--cache-data` implies `--cache-metadata`.

Reply:
If we are starting with a single flag, would it make sense to have something like `--cache <DIR>`, which enables both metadata and data cache using `<DIR>` for the latter?

We can always add more specific flags like `--cache-metadata` and `--cache-data <DIR>` if we want to give users more detailed options, without breaking changes or ambiguities.

Reply:
I updated to use `--cache <DIR>`. We should in a separate PR make an update to the final CLI args which also adopts this (or makes whatever change we decide to make).